fdbserver/VersionedBTree.actor.cpp

```diff
@@ -3431,18 +3431,19 @@ class DWALPager final : public IPager2 {
 		state Version oldestRetainedVersion = self->effectiveOldestVersion();
 
 		// Cutoff is the version we can pop to
-		state RemappedPage cutoff(oldestRetainedVersion - self->remapCleanupWindow);
-		// Minimum version we must pop to before obeying stop command.
-		state Version minStopVersion =
-		    cutoff.version - (BUGGIFY ? deterministicRandom()->randomInt(0, 10)
-		                              : (self->remapCleanupWindow * SERVER_KNOBS->REDWOOD_REMAP_CLEANUP_LAG));
-
-		debug_printf("DWALPager(%s) remapCleanup cutoff.version %" PRId64 " oldestRetainedVersion=%" PRId64
-		             " minStopVersion %" PRId64 " items=%" PRId64 "\n",
+		state RemappedPage cutoff(oldestRetainedVersion);
+
+		// Maximum number of remaining remap entries to keep before obeying stop command.
+		state uint64_t maxRemapEntries =
+		    self->remapCleanupWindow *
+		    (1 + (BUGGIFY ? deterministicRandom()->randomInt(0, 20) / 100.0 : SERVER_KNOBS->REDWOOD_REMAP_CLEANUP_LAG));
```

Comment:
@sfc-gh-satherton I forget about what did you suggest for the randomized value when buggify. Let me know if this doesn't look right.